thirdparty/github.go

```diff
@@ -60,72 +60,110 @@ var (
 	GithubPatchBaseRepoKey  = bsonutil.MustHaveTag(GithubPatch{}, "BaseRepo")
 )
 
-func githubShouldRetry(index int, req *http.Request, resp *http.Response, err error) bool {
-	if index >= NumGithubAttempts {
-		return false
-	}
+func githubShouldRetry(caller string) utility.HTTPRetryFunction {
+	return func(index int, req *http.Request, resp *http.Response, err error) bool {
+		if index >= NumGithubAttempts {
+			return false
+		}
 
-	url := req.URL.String()
+		url := req.URL.String()
 
-	if err != nil {
-		temporary := utility.IsTemporaryError(err)
-		grip.Error(message.WrapError(err, message.Fields{
-			"message":   "failed trying to call github",
-			"method":    req.Method,
-			"url":       url,
-			"temporary": temporary,
-		}))
-		return temporary
-	}
+		if err != nil {
+			temporary := utility.IsTemporaryError(err)
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":   "failed trying to call github",
+				"method":    req.Method,
+				"url":       url,
+				"temporary": temporary,
+			}))
+			if temporary {
+				grip.Info(message.Fields{
+					"ticket":    "EVG-14603",
+					"message":   "error is temporary",
+					"caller":    caller,
+					"retry_num": index,
+				})
+			}
+			return temporary
+		}
 
-	if resp == nil {
-		return true
-	}
+		if resp == nil {
+			grip.Info(message.Fields{
+				"ticket":    "EVG-14603",
```

Comment:
[EVG-14603](https://jira.mongodb.org/browse/EVG-14603) is already closed. Is this supposed to be EVG-17069?

Reply:
I'm neutral about which ticket number we use. I don't think we're ever going to want to remove this logging but it's helpful to have a common thread for investigating. I can change this to something neutral.